### PR TITLE
MULE-17828: fix tests

### PR DESCRIPTION
**Pull request by marianogonzalez**

### CODE REVIEW

File: core-tests/src/test/java/org/mule/runtime/core/internal/el/dataweave/DataWeaveExpressionLanguageAdaptorTestCase.java [hunk 1]
```diff
@@ -649,7 +649,8 @@ public void compilationExceptionPropagates() {
   @Test
   public void evaluateInvalidCompiledExpression() throws MuleException {
     ExpressionCompilationException e = new ExpressionCompilationException(createStaticMessage("oopsy"));
-    expectedEx.expect(is(sameInstance(e)));
+    expectedEx.expect(ExpressionRuntimeException.class);
```
Comment (elrodro83) [inline core-tests/src/test/java/org/mule/runtime/core/internal/el/dataweave/DataWeaveExpressionLanguageAdaptorTestCase.java]: we should assert thet the error message is meaningful here

File: core/src/main/java/org/mule/runtime/core/internal/processor/simple/ParseTemplateProcessor.java [hunk 2]
```diff
@@ -112,6 +116,8 @@ private MediaType buildMediaType() {
 
   @Override
   public CoreEvent process(CoreEvent event) {
+    evaluateCorrectArguments();
```
Comment (elrodro83) [inline core/src/main/java/org/mule/runtime/core/internal/processor/simple/ParseTemplateProcessor.java]: can't this be checked on initialize? or does it cause problems

Comment (marianogonzalez) [reply core/src/main/java/org/mule/runtime/core/internal/processor/simple/ParseTemplateProcessor.java]: it breaks backwards compatibility. THat's why the test broke

File: core-tests/src/test/java/org/mule/runtime/core/internal/el/dataweave/DataWeaveExpressionLanguageAdaptorTestCase.java (hunk 1 above)
Comment (marianogonzalez) [reply core-tests/src/test/java/org/mule/runtime/core/internal/el/dataweave/DataWeaveExpressionLanguageAdaptorTestCase.java]: no point in doing that. I'm asserting that the thrown assertion is the same as in the mock

**Comment by marianogonzalez** [comment]
--validate

**Pull request merged**